apps/files/src/components/FilesLists/StatusIndicators/DefaultIndicators.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <div>
+    <oc-button
+      v-for="(indicator, index) in indicators"
+      :key="index"
+      class="file-row-share-indicator uk-text-middle"
+      :class="{ 'uk-margin-xsmall-left' : index > 0 }"
+      :aria-label="indicator.label"
+      @click="indicator.handler(item, indicator.id)"
+      variation="raw"
+    >
+      <oc-icon
+        :name="indicator.icon"
+        class="uk-text-middle"
+        size="small"
+        :variation="indicator.status"
+      />
+    </oc-button>
+  </div>
+</template>
+
+<script>
+import intersection from 'lodash/intersection'
+import { shareTypes } from '../../../helpers/shareTypes'
+import { getParentPaths } from '../../../helpers/path'
+
+const userShareTypes = [shareTypes.user, shareTypes.group, shareTypes.guest, shareTypes.remote]
+
+export default {
+  name: 'StatusIndicators',
+
+  props: {
+    item: {
+      type: Object,
+      required: true
+    },
+    currentFolderPath: {
+      type: String,
+      required: true
+    }
+  },
+
+  computed: {
+    indicators () {
+      const indicators = []
+
+      if (this.isUserShare(this.item)) {
+        indicators.push({
+          id: 'files-sharing',
+          label: this.shareUserIconLabel(this.item),
+          icon: 'group',
+          status: this.shareUserIconVariation(this.item),
+          handler: this.indicatorHandler
+        })
+      }
+
+      if (this.isLinkShare(this.item)) {
+        indicators.push({
+          id: 'file-link',
+          label: this.shareLinkIconLabel(this.item),
+          icon: 'link',
+          status: this.shareLinkIconVariation(this.item),
+          handler: this.indicatorHandler
+        })
+      }
+
+      return indicators
+    },
+
+    shareTypesIndirect () {
```

Comment:
why not keep all these methods private with `$_` prefix ?

Reply:
Haven't seen a reason to keep them as private since it's encapsulated from anything else.